lib/web/desktop.go

```diff
@@ -392,6 +392,33 @@ func proxyWebsocketConn(ws *websocket.Conn, wds net.Conn) error {
 	return trace.NewAggregate(retErrs...)
 }
 
+// wasWebsocketClosedByClient handles the error returned by proxyWebsocketConn
+func handleProxyWebsocketConnErr(err error, log *logrus.Entry) {
+	if err == nil {
+		return
+	}
+
+	if agg, ok := trace.Unwrap(err).(trace.Aggregate); ok {
+		wsClosedByClient := false
+		for _, err := range agg.Errors() {
+			if ce, ok := trace.Unwrap(err).(*websocket.CloseError); ok {
+				switch ce.Code {
+				case websocket.CloseNoStatusReceived, // when the user hits "disconnect" from the menu
```

Comment:
Should we update the UI to actually send a close when the user hits disconnect in the menu?

Reply:
846afbe85ad0fda356857743a2346a1ddb534877 + https://github.com/gravitational/webapps/pull/1442